src/base/memory_monitor_sr.cpp

```
@@ -557,16 +694,26 @@ void mmon_resize_stat (THREAD_ENTRY *thread_p, MMON_STAT_ID stat_id, int64_t old
     }
 }
 
-int mmon_aggregate_module_info (MMON_MODULE_INFO *info, int module_index)
+int mmon_aggregate_module_info (MMON_MODULE_INFO &info, int module_index)
 {
   int error = NO_ERROR;
 
+  if (mmon_Gl != nullptr)
+    {
+      error = mmon_Gl->aggregate_module_info (info, module_index);
+    }
+
   return error;
 }
 
-int mmon_aggregate_tran_info (MMON_TRAN_INFO *info, int tran_count)
+int mmon_aggregate_tran_info (MMON_TRAN_INFO &info, int tran_count)
 {
   int error = NO_ERROR;
 
+  if (mmon_Gl != nullptr)
```

I think it must not be called if memmon is not initialized. It has to be handled as an error on the upperside or set an error here. So, `assert (mmon_Gl != nullptr)` or it should return an error. `NO_ERROR` with the empty `info` seems bad.

---

I agree. I change it to assertion.